fabric-fluids-v1/src/main/java/net/fabricmc/fabric/api/fluids/FluidView.java

```diff
@@ -0,0 +1,125 @@
+package net.fabricmc.fabric.api.fluids;
+
+import java.util.Collections;
+import java.util.IdentityHashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Iterators;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.Entity;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Box;
+import net.minecraft.util.math.Direction;
+import net.minecraft.world.World;
+
+import net.fabricmc.fabric.api.fluids.containers.FluidContainer;
+import net.fabricmc.fabric.api.fluids.containers.SimpleFluidContainer;
+import net.fabricmc.fabric.api.fluids.containers.volume.ImmutableFluidVolume;
+import net.fabricmc.fabric.api.fluids.items.ItemFluidContainer;
+import net.fabricmc.fabric.api.fluids.items.ItemSink;
+import net.fabricmc.fabric.api.fluids.math.Drops;
+import net.fabricmc.fabric.api.fluids.minecraft.items.BottleItemFluidContainer;
+import net.fabricmc.fabric.api.fluids.minecraft.items.BowlItemFluidContainer;
+import net.fabricmc.fabric.api.fluids.world.SidedFluidContainer;
+
+/**
+ * A util class for accessing fluid containers
+ * <p>
+ * Introduction:
+ *
+ * @see Drops
+ */
+public class FluidView {
+	public static final Map<Item, ItemFluidContainer> FLUID_CONTAINERS = new IdentityHashMap<>();
+
+	static {
+		FLUID_CONTAINERS.put(Items.BOWL, ((waste, stack) -> new BowlItemFluidContainer(stack, waste)));
+		FLUID_CONTAINERS.put(Items.DRAGON_BREATH, ((waste, stack) -> new BottleItemFluidContainer(stack, waste)));
+	}
+
+	private FluidView() {
+	}
+
+	public static FluidContainer getFluidContainer(ItemStack stack, ItemSink sink) {
+		Item item = stack.getItem();
+		if (item instanceof ItemFluidContainer) {
+			return ((ItemFluidContainer) item).getContainer(sink, stack);
+		}
+		ItemFluidContainer container = FLUID_CONTAINERS.get(item);
+		if (container != null) {
+			return container.getContainer(sink, stack);
+		}
+		return ImmutableFluidVolume.EMPTY;
+	}
+
+	/**
+	 * merge all the entity and block containers in the given location for the given face
+	 * into one fluid container
+	 *
+	 * @param world the world
+	 * @param pos the position
+	 * @param face the face
+	 * @return a fluid container for the face
+	 */
+	public static FluidContainer getContainer(World world, BlockPos pos, Direction face) {
```

Comment:
Why is this called `getContainer` while the previous method is `getFluidContainer`

Reply:
good point, I'm not actually happy with most of the names I use